lib/mqtt/mqtt.dart

```diff
@@ -0,0 +1,92 @@
+library mqtt;
+
+import 'dart:async';
+
+import 'package:ltbl/util/stream_extensions.dart';
+import 'package:mqtt5_client/mqtt5_client.dart';
+import 'package:mqtt5_client/mqtt5_server_client.dart';
+import 'package:typed_data/typed_buffers.dart';
+
+part './subscription.dart';
+
+typedef ReceivedMessage = MqttReceivedMessage<MqttMessage>;
+
+class MqttStreamClient {
+  late final MqttStreamSubscriptionManager _subscriptionManager;
+  final StreamController<MqttConnectionStatus> _connectionStatusController =
+      StreamController.broadcast();
+  late final Stream<MqttConnectionStatus> connectionStatus;
+  final MqttClient _client;
+  final String? _username;
+  final String? _password;
+
+  MqttStreamClient(
+    String serverAddress,
+    String uniqueID,
+    int port, [
+    String? username,
+    String? password,
+  ])  : _client = MqttServerClient.withPort(serverAddress, uniqueID, port),
+        _username = username,
+        _password = password {
+    _client.autoReconnect = true;
+    _client.resubscribeOnAutoReconnect = true;
+
+    // we need to connect before the updates and the published streams
+    // are valid, so just forcibly connect here
```

Comment:
this comment is a bit unclear
```suggestion
    // we need call connect for the updates and the published streams
    // to be valid, so just forcibly connect here to be able
    // to listen to them already in the constructor
```